### PR TITLE
Optimize CI a bit

### PR DESCRIPTION
**Pull request by Goooler**
### :pencil: Description
Mainly for using gradle-build-action to speed up build on CI, and bump action versions by the way.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -14,30 +14,25 @@ jobs:
...
           distribution: 'zulu'
 
       - name: Set up Gradle cache
-        uses: actions/cache@v1
-        with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle.kts') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-
+        uses: gradle/gradle-build-action@v2
```
Comment (Goooler) [inline .github/workflows/continuous-integration.yml:29]: https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.

File: .github/workflows/release-code.yml [hunk 2]
```diff
@@ -11,16 +11,16 @@ jobs:
...
       - uses: gradle/wrapper-validation-action@v1
 
       - name: Set up Java 11
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
         with:
           java-version: 11
           distribution: 'zulu'
-
+      - uses: gradle/gradle-build-action@v2
```
Comment (dariuszkuc) [inline .github/workflows/release-code.yml:23]: lets remove it -> when releasing the code we want to ensure whole thing can be build without relying on any caching

Comment (Goooler) [reply .github/workflows/release-code.yml]: 5726ea9021bacf1ebea9b707e744ad7c4f9c47e2

**Review by dariuszkuc — APPROVED** [review]

**Comment by Goooler** [comment]
```
Error:  Failed to execute goal com.expediagroup:graphql-kotlin-maven-plugin:6.0.0-SNAPSHOT:generate-client (default) on project graphql-kotlin-maven-client-example: Execution default of goal com.expediagroup:graphql-kotlin-maven-plugin:6.0.0-SNAPSHOT:generate-client failed: A required class was missing while executing com.expediagroup:graphql-kotlin-maven-plugin:6.0.0-SNAPSHOT:generate-client: com/expediagroup/graphql/client/Generated
Error:  -----------------------------------------------------
Error:  realm =    plugin>com.expediagroup:graphql-kotlin-maven-plugin:6.0.0-SNAPSHOT
Error:  strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
```

**Comment by dariuszkuc** [comment]
Looks like the `gradle-build-action` does not play nicely with caching action for Maven (we publish both Maven and Gradle plugins and need to run Maven for plugin integration tests).

**Comment by Goooler** [comment]
Maybe it's triggered by this?
```
Error: [ERROR] Failed to execute goal com.expediagroup:graphql-kotlin-maven-plugin:6.0.0-SNAPSHOT:introspect-schema (default) on project graphql-kotlin-introspect-schema-mojo-timeout-test: 
Error: [ERROR] Execution default of goal com.expediagroup:graphql-kotlin-maven-plugin:6.0.0-SNAPSHOT:introspect-schema failed: Request timeout has expired [url=http://localhost:44333/graphqlTimeout, request_timeout=100 ms] -> [Help 1]
Error: [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
Error: [ERROR] Re-run Maven using the -X switch to enable full debug logging.
Error: [ERROR] 
Error: [ERROR] For more information about the errors and possible solutions, please read the following articles:
Error: [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```
I've tested again on my repo, it passed. See https://github.com/Goooler/graphql-kotlin/runs/5605041916?check_suite_focus=true

File: .github/workflows/pr-check.yml [hunk 3]
```diff
@@ -15,28 +15,23 @@ jobs:
...
           java-version: 11
           distribution: 'zulu'
 
       - name: Set up Gradle cache
-        uses: actions/cache@v1
-        with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle.kts') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-
```
Comment (Goooler) [inline .github/workflows/pr-check.yml:35]: `gradle/gradle-build-action` just caches the `.gradle` folder, `.m2` folder is handled by [actions/cache](https://github.com/Goooler/graphql-kotlin/blob/b3ae3449dbe968e576406bab5632244dbf92d600/.github/workflows/pr-check.yml#L34-L39).

**Pull request merged**